### ISSUE TITLE
retroPoints

### ISSUE THREAD

**Issue by thebluecorsair** [Retrospring/bugs#54]
**What is retroPoints?**
retroPoints is a new system for Retrospring that rewards users with points for activeness and quality content. Points can be used to get special features on the service including but not limited to:
* Appear as a "featured" user on the Discover page.
* Have a question become featured for 24 hours, which will appear pinned to the top of all user's Inbox. (All questions are reviewed by the Retrospring team before accepted.)
* Badges (discussed below).

**How can I earn retroPoints?**
Earning points on Retrospring is very easy and is designed to make earning rewards and unlocks enjoyable. Most "obtainables" are made to be able to be earned after two to three days of activeness. The following will make a user gain one point:
* Receiving an answer to one of your questions.
* Receiving a smile on an answer from a user.
* Accessing Retrospring daily (1 point per day)
* More ways coming soon.

**Badges**
Certain badges will be able to be purchased with points and are permanently added to the account. Badges allow users to show their involvement with the community and give another purpose of staying active. Some badges will only be offered for set periods of time, for special events and holidays, and so forth. This gives an incentive for users to save up points to unlock badges that may not be available ever again to show off to their friends.

**Can this be abused?**
Yes and no. The system will prevent purposeful point generating from many different angles, including spamming questions for the intent of generating points. The Retrospring team will have the ability to disable the gaining of points for a time period of a day, week, or month when necessary.

**Want an easier way to gain points?**
Purchase a retroStim! This will double your points for an entire month, starting from the time or purchase. This greatly assists with those who wish to collect badges or just get their name out by using the available rewards.

Please share your feedback and thoughts below!

[Note: This is not a final specification, and will be edited prior to public release.]

**Comment by TheHowl** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA256

Also, you can buy retroPoints at the exchange rate of 100$ for one, if
you really want to get a lot of them quickly!

On 05/14/2015 12:28 AM, The Blue Corsair wrote:
> *What is retroPoints?* retroPoints is a new system for Retrospring
> that rewards users with points for activeness and quality content.
> Points can be used to get special features on the service including
> but not limited to:
> 
> * Appear as a "featured" user on the Discover page. * Have a
> question become featured for 24 hours, which will appear pinned to
> the top of all user's Inbox. (All questions are reviewed by the
> Retrospring team before accepted.) * Badges (discussed below).
> 
> *How can I earn retroPoints?* Earning points on Retrospring is very
> easy and is designed to make earning rewards and unlocks enjoyable.
> Most "obtainables" are made to be able to be earned after two to
> three days of activeness. The following will make a user gain one
> point:
> 
> * Receiving an answer to one of your questions. * Receiving a smile
> on an answer from a user. * Accessing Retrospring daily (1 point
> per day) * More ways coming soon.
> 
> *Badges* Certain badges will be able to be purchased with points
> and are permanently added to the account. Badges allow users to
> show their involvement with the community and give another purpose
> of staying active. Some badges will only be offered for set periods
> of time, for special events and holidays, and so forth. This gives
> an incentive for users to save up points to unlock badges that may
> not be available ever again to show off to their friends.
> 
> *Can this be abused?* Yes and no. The system will prevent
> purposeful point generating from many different angles, including
> spamming questions for the intent of generating points. The
> Retrospring team will have the ability to disable the gaining of
> points for a time period of a day, week, or month when necessary.
> 
> *Want an easier way to gain points?* Purchase a retroStim! This
> will double your points for an entire month, starting from the time
> or purchase. This greatly assists with those who wish to collect
> badges or just get their name out by using the available rewards.
> 
> Please share your feedback and thoughts below!
> 
> [Note: This may be edited from time to time until it has been
> officially implemented.]
> 
> — Reply to this email directly or view it on GitHub 
> <https://github.com/Retrospring/bugs/issues/54>.
> 
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v2

iQIcBAEBCAAGBQJVVDFdAAoJEFCVSR4ynJMlNJIP/3DnWUiFPJcnbUP4fZp8A+ec
v6i3t+3qhIQRMhk1Vp6YbLhz+uYVEAhz2/DfjCR7sD+L6VIczUq2m/3UdIKgID29
ItyrRmdezJHvGrD8hSmTuRYBWXAFKten16QyClFiGWsgiVLUGQJmYrX8WwBr2mkG
P4Xdme4LzKoiWHxYFs2Tss82MGeUdeps5uVBzH1F4/aKYXVVyL1uj+LZG8hSyQo5
hJ2WbXNEPsIx6d3oB3kDpcV8qoW5CZT8GrD2fsd71cAmq1Kci7aJDpj56/Iu82DT
BXVPs3G7myHD5+q/JcUYO5XlRyWcUlpf91LphFUh7nwnNOdgMuos1xnLFNdCFabS
7cVSW8Ky3FEabqemGEqe4zRHKeUu7LiYdqH/RGiVaxpHX/w2cxbOGwdQx8xMhEs+
sME52FH1biReDXtVgw3nX5y+1Rofrd0paoMpcgwBrkU/OM1s9ksS1k1/Reb8uXbS
uCBcBSQsch86RgdhXNcHJxkV7aPMBk5C+6Qt25VDeqvt3IadF/Li8tFwaVaM/goX
RY4ExRff1UgmwjWs3Yup5S+saIPhOO6RbWSjB4EQ/nsHXEAYgmoRHYkc34K1k/b5
0MgbwEkrlS+hDQaC+o1UGsG3/T+P2gX+J1fsttwAPOPIYANg3za3e4mig0KbZu6w
hfPvrZdWZnCQtWAJX2NP
=+sjB
-----END PGP SIGNATURE-----

**Comment by thebluecorsair** [comment]
@TheHowl Pls. $1 = π retroPoints.
(but in all seriousness, avoiding any direct $ to points system through discussions, as it just negatively impacts the point system and doesn't encourage activeness.)

**Comment by yukimono** [comment]
So I was talking about the Apple iOS fees over in ``slack`` and i was thinking
how about we serve ads, and then hide the ads for users that have *n* RP?

**Comment by TheHowl** [comment]
this is a joke right

On 14 May 2015 23:11:48 CEST, Yuki <notifications@github.com> wrote:
>So I was talking about the Apple iOS fees over in ``slack`` and i was
>thinking
>how about we serve ads, and then hide the ads for users that have *n*
>RP?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/Retrospring/bugs/issues/54#issuecomment-102170608

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by thebluecorsair** [comment]
@TheHowl It's actually being considered and worked on. :hamburger:

**Comment by TheHowl** [comment]
okay, i'll be serious now

although i dislike a bit retrostim and would rather prefer a donation system if money's the problem, i'm completely against ads on retrospring. this wouldn't make retrospring different from any other qa website, and the fact that with x retropoints you can disable them is just stupid imho. it would make retrospring be just like any other mobile game where there are coins and gold. with gold you can boost the coins and with coins you can buy some stuff. there are also ads you can disable by using some gold. the difference on retrospring is that you use coins to disable ads.

tl;dr: this idea is cool, but please no ads

On 14 May 2015 23:36:32 CEST, The Blue Corsair <notifications@github.com> wrote:
>@TheHowl It's actually being considered and worked on. :hamburger: 
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/Retrospring/bugs/issues/54#issuecomment-102177371

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by thebluecorsair** [comment]
@TheHowl You're in luck as I think a majority (if not all) of the team all dislike and are against the idea of ads, and thus they will likely not be added to the service. Instead, we will likely be bringing back the Patreon and offer tiers focused around retroPoints and build upon it.

This system is not intended to add any form of restrictions to users in any way, but instead offers a way and an incentive towards being active and earning new things because of it. :+1:

**Comment by thebluecorsair** [comment]
**More Information About retroPoint Gain**
The scaleability of retroPoint rewards will be monitored and adjusted in accordance to expected daily gain. Here is a good example of the expected daily gain of an active user.

If a user asks five questions in a day, and each question gets twenty answers, the user is given 100 points. If the user answers ten questions and gets a smile on each, the user would be given another 10 points.

For a retroStim, it is expected to take 12-15 days to obtain.

Purchasing multiple retroSim will simply raise the duration. The maximum duration is currently set to 60 days in the case of retroSim price changes in the future.

**Changes**
– Purchasing a retroSim through the service will cost an expected **1500 points** for a limited time. This will then raise to **3000-4500 points** later on.
– Appearing as a "featured" user on the Discover page for 24 hours will cost an expected **550-750 points**. When there are multiple users featured, they will appear as a carousel and switch between all users every few seconds.
– Having a question appear as featured for 6 hours will cost an expected **350-550 points**. They will appear for 6 hours upon approval. There is no longer a 24 hour option.

**The Re-introduction of Patreon and Updated Tiers**
To better assist the current Retrospring administration team in providing the service to us all, we will be re-introducing Patreon as a means of providing funding. Instead of the rewards offered before, we will be relaunching it with the following tiers:
* **$3.50** – All donations over 3.50$ are rewarded with a Supporter badge. This is permanently bound to an account, even if you stop the donations. You are also given a retroStim that doubles experience given across your account for the duration of the Patreon donation. *More information about retroStim can be found above*.
* **$7.50** – This provides all incentives of the previous tier, as well as early access to new features before they are released publicly.

More tiers to come.

**Comment by thebluecorsair** [comment]
[please note, this post has been heavily edited since first posted.]

**Levels**
The team has actively been discussing implementing a Level system to Retrospring that would work alongside retroPoints in terms of unlocks. As points are gained, a user's level progression increases, unlocking certain benefits to the service. These benefits are meant to encourage activity and not to interfere with the average user's experience.

As brought up in a concern by a Github user ( http://pastebin.com/nyw3Urws ), we are avoiding the mentality of Reddit in which posts are crafted just to gain karma/points. To work around this, the amount of points a user possesses is not publicized nor shown to the user. Instead, only a player's Level (and a circular progress bar displaying progress to the next level is displayed.

Additionally, I have reworked the entire point system which can be found below.

**Extended Information**
The scaleability of retroPoint rewards will be monitored and adjusted in accordance to expected daily gain. Any changes to the system will always be presented publicly through the Retrospring blog.

**Changes**
– retroStims are only obtainable through the means of donating through Patreon and are limited by the duration of the donation period. Upon reaching certain Levels (discussed below), temporary retroStims are given for a short duration of time. retroStims can not be directly purchased with points to prevent unbalanced gain of retroPoints.
– A level system will be implemented.

**Levels and Unlocks**
* Level 1 – Initial level, no benefits.
* Level 5 – Unlocks the ability to post a single featured question once a week. This question is sent to a approval queue and must be approved by appointed team members.
* Level 7 – Unlocks the ability to post an additional featured question once a week (Total: 2).
* Level 15 – Unlocks the ability to become a "featured" member on the Discover page for 12 hours once a week, as well as a 7-day retroStim. Unlocks first badge.
* Level 30 – Unlocks a badge.
* Level 50 – Unlocks a badge, and an additional featured question once a week (Total: 3).
* Level 100 – Highest level, unlocks a badge. [more will be planned later]

The points required for each level will be planned later.

**The Re-introduction of Patreon and Updated Tiers**
To better assist the current Retrospring administration team in providing the service to us all, we will be re-introducing Patreon as a means of providing funding. Instead of the rewards offered before, we will be relaunching it with the following tiers:
* **$3.50** – All donations over 3.50$ are rewarded with a Supporter badge. This is permanently bound to an account, even if you stop the donations. You are also given a retroStim that doubles experience given across your account for the duration of the Patreon donation. *More information about retroStim can be found above*. For June 2015, you will be given 300 additional points upon donating.
* **$7.50** – This provides all incentives of the previous tier, as well as early access to new features before they are released publicly. For June 2015, you will be given 200 additional points upon donating. (500 total).

More tiers to come.

**Comment by TheCoward** [comment]
Please, is this really necessary? I mean, it's like making this whole thing a DLC whore. As a user, I don't really see the point of this, and most other users will also think it's a useless gadget like the badges on Ask.fm. Sorry to point that out so harshly, but that's how I feel about it.

**Comment by calvcoll** [comment]
I feel the same as @TheCoward, but to me it just seems like a feature stuck on, but it would catch the people who are influenced by freemium aspects of products, I'd personally prefer supporting via donations/patreon.

**Comment by yukimono** [comment]
I'm personally indifferent on the concept, since we're on patreon i see no reason to implement this as of yet, however at some point a similar system may or may not be implemented for different reasons, mostly self promoting with the recently added Discovery page.

**Comment by TheCoward** [comment]
But yes, the issue is, it's going to become like pay2win video games (I think the name speaks for itself) and those who can pay to get their content forwarded will do and others will be stuck behind. Either way, I've seen how much people hate promoted tweets AND ads on Twitter, so see?
I don't think you should do that, because "It's like you're throwing the greatest party on campus and someone saying it's gotta be over by eleven."
My opinion is that you should indeed let patrons get access to the Alpha testing environment, and custom badges, which put their answers forward, but other than that, not cool.
Then again it's just my opinion.

**Comment by TheHowl** [comment]
Then again, it's just your opinion, but I (and I think a lot of other 
users as well) agree with you.

Il 09/06/2015 11:20, Scimitar ha scritto:
> e is, it's going to become like pay2win video games (I think the name 
> speaks for itself) and those who can pay to get their content 
> forwarded will do and others will be stuck behind. Either way, I've 
> seen how much people hate promoted tweets AND ads on Twitter, so see?
> I don't think you should do that, because "It's like you're throwing 
> the greatest party on campus and someone saying it's gotta be over by 
> eleven."
> My opinion is that you should indeed let patrons get access to the 
> Alpha testing environment, and custom badges, which put their answers 
> forward, but other than that, not cool.
> Then again it's just my opinion.

**Comment by yukimono** [comment]
Funny thing about promoted and ad tweets

i've never seen one.

**Comment by yukimono** [comment]
At the end of the day a pure and free service won't survive for that long especially with the growth Retrospring has, it costs money to keep the server up and running and with the more people using the site the more expensive it's eventually going to get, especially with content hosting (profile images/headers) and the search engine, app store licenses also cost a few per year.

Having that said, Retrospring is on Patreon and as long as that is working out, the finance of the service is sane enough that we aren't burning money and I see no need for a concept such as RetroPoints.

tl;dr as long as people keep supporting RS via Patreon, we won't need to add counter intuitive measures to keep the server from making everyone involved bankrupt.

**Comment by TheHowl** [comment]
tweetdeck doesn't show them, as well as pretty much all non-official 
twitter clients.
Twitter web and android at least use them.

Il 09/06/2015 11:23, Yuki ha scritto:

**Comment by yukimono** [comment]
Last time I checked AWS costs about $8/mo, assuming RS runs on a standard VPS node it costs at around $10/month

For the sake of easy math it's $20/mo, $240/yr. adding $10/yr for the .net domain (From the registrar listed on the whois record), and the upcoming app which will add $100/yr.

meaning it'll be around $350/yr when search and mobile apps are added.

**Comment by yukimono** [comment]
And I'm not saying that we can't afford that right now, we can. Thanks to Patreon, we can.
I'm just saying, this is "Plan B" (to me) when Patreon doesn't work.

**Comment by yukimono** [comment]
***I removed the comment about the cost of the site because I don't actually know the figures and I didn't want to make scenareos based on assumed numbers, you have to poke Nilsding about how much it costs to run Retrospring***

**Comment by TheHowl** [comment]
$60/mo. ezpz. It's written on the patreon.

Il 09/06/2015 11:43, Yuki ha scritto:
> */about the cost of the site because I don't actually know the figures 
> and I didn't want to make scenareos based on assumed numbers, you have 
> to poke Nilsding about how much it costs to run Retrospring/*

**Comment by yukimono** [comment]
Yes, I know. That's what I'm saying, my comment earlier clearly stated *"since we're on patreon i see no reason to implement this as of yet"* and then a few comments later I explained what I meant with *"we __may or may not__ implement this later"*

**Comment by yukimono** [comment]
To be perfectly clear, I have no intention of adding this in the foreseeable future, which is why i haven't tagged this with a milestone like I did with most other requests/issues. As far as I can tell neither do Nilsding or Pixeldesu. ***This is just a concept floating around which will probably be forgotten and never come to fruition because of the awesome community everyone helped us build with Retrospring that is keeping the site alive.***

**Comment by TheHowl** [comment]
>At the end of the day a pure and free service won't survive for that 
long especially with the growth Retrospring has, it costs money to keep 
the server up and running and with the more people using the site the 
more expensive it's eventually going to get, especially with content 
hosting (profile images/headers) and the search engine, app store 
licenses also cost a few per year.

Not true.

As a true weeaboo, I'd like to take in this discussion 
[osu!](http://osu.ppy.sh). osu! is free and ad-free, and not 
pay2win/play/whatever. It is completely donation-based. Yet all this is 
a fact:

* Huge userbase (6,414,720 as of now, probably about ~35,000 active per day)
* Lots of data to store (at least ~125000 beatmaps on the website to 
store, all beatmaps contain at least an mp3 and a background = 5mb on 
average, 610GB at least to store on an S3, that needs as well to 
replicate on other three mirrors - also the database is huge and in 
continuous growth, as every score takes, say, 0,25kBytes in the 
database, multiply that number for 3,030,000,000 (3,03 billion, as the 
website states), you get 711GB)
* Some of the developers don't actually just do it for free! At the very 
least, peppy's salary is from osu!, and living in Japan ain't cheap. 
Also there are a few full-time developers/community managers, so we can 
take that up to at least 3 salaries, as well as a few part-time ones (I 
don't think flyte is doing [all this](http://next.ppy.sh) for free).

So in a few words, roughly:

* $4000 at least goes into paying the staff
* $1000 at least goes for servers
* Probably $400 go into a lot of other services (Slack, GitHub, y'know)

$5400 /at least/ are spent /every month/. But everything is supported by 
volountary $4/mo donations - of course with a donation there are some 
goodies, but it still can't really be considered "premium" imho.

(P.s.: Yeah, I've read your last message now, but I started writing this 
~4 messages before, and I'm sending this one anyway even if you wrote 
that you're going to implement this /in extremis/.)

**Comment by yukimono** [comment]
nope, you're wrong
https://osu.ppy.sh/p/support

**Comment by TheHowl** [comment]
Tell me exactly were I was wrong. I **DID** mention that.

> It is completely donation-based.
>But everything is supported by volountary $4/mo donations - **of course with a donation there are some goodies, but it still can't really be considered "premium" imho.**

**Comment by yukimono** [comment]
wait nvm i read that last paragraph.

but yeah, that model Osu! currently has is what we also currently have, and it's not purely free, people are paying for it indirectly through support (osu!) or patreon (retrospring)

**Comment by yukimono** [comment]
*purely free* means (to me) that nobody using the site is spending anything but the bandwidth costs to their ISP.

**Comment by yukimono** [comment]
also i lol'd

one of the benefits of supporting Osu! is "No ads anywhere. Unlike 99.95% of the web, we don't profit off shoving stuff in your face.", which is something I mentioned waaaaaay earlier, before we had a patreon

**Comment by yukimono** [comment]
i should sleep before saying anything more.

**Comment by TheHowl** [comment]
Well, then I think the thing is we have different definitions of *purely free*. imho it is when you *don't have to* pay anything, but you *can* support, although support doesn't mean *premium*, as all the core features still remain free even without any support.

**Comment by yukimono** [comment]
yes, which is also what corsair outlined, the core features were untouched, infact everything you'd gain from this is something we'd have to newly implement, we have no featured system.

**Comment by yukimono** [comment]
i'm also 100% sure this started off as a joke in slack

**Comment by yukimono** [comment]
If you look at Osu!'s support page.
https://osu.ppy.sh/p/support

some of the benefits are borderline freemuim.

> Customise your profile by adding a fully customisable user page.

What if we added user theming, exclusive to patreon supporters?

> More lenient download restrictions, especially when using osu!direct.

What if we added a profile image/header size limit, but it would be a lot less strict with patreon supporters?

> The ability to change your username (once max).

What if we did this?

**Comment by yukimono** [comment]
Closing issue because motive for retroPoints is handled with Patreon.

**Comment by TheHowl** [comment]
>More lenient download restrictions, especially when using osu!direct.
>What if we added a profile image/header size limit, but it would be a lot less strict with patreon supporters?

This is outdated. Download restriction have been disabled as of a few months ago. Supporters __might__ still get a few kB/s more, but not that much of a big deal.

>The ability to change your username (once max).
>What if we did this?

The truth is, it originally wasn't advertised as "FEATURE SUPPORTER-ONLY", the supporter-exclusive name change was just a method to avoid users that just signed up on osu! and already want to change their name. A sort of cooldown.

>Customise your profile by adding a fully customisable user page.
>What if we added user theming, exclusive to patreon supporters?

I might agree on this.

Also, lol

![image](https://cloud.githubusercontent.com/assets/4681308/8055629/37106508-0ea1-11e5-8ba9-9f5e4d7e1b80.png)

**Comment by yukimono** [comment]
"orz sorry for the spam

**Comment by TheCoward** [comment]
>tweetdeck doesn't show them, as well as pretty much all non-official
twitter clients.
Unfortunately, I believe that a while ago, they were pushing updates that were transitioning towards the implementation of ads in TweetDeck. Don't know if it's still true as of today, and I hope not. Either way, I hope promoted tweets will have a special CSS identifier, because that way it's as simple as applying CSS magic, aka 
```css
display: none
```

**Comment by TheCoward** [comment]
Also I dislike osu! so that argument doesn't really strike me as a defense, rather the opposite.
